python/ray/tests/test_state_api_log.py

```diff
@@ -585,25 +585,27 @@ def getpid(self):
 
 def test_log_list(ray_start_cluster):
     cluster = ray_start_cluster
-    cluster.add_node(num_cpus=0)
+    num_nodes = 5
+    for _ in range(num_nodes):
+        cluster.add_node(num_cpus=0)
     ray.init(address=cluster.address)
 
     def verify():
-        head_node = list_nodes()[0]
-        # When glob filter is not provided, it should provide all logs
-        logs = list_logs(node_id=head_node["node_id"])
-        assert "raylet" in logs
-        assert "gcs_server" in logs
-        assert "dashboard" in logs
-        assert "agent" in logs
-        assert "internal" in logs
-        assert "driver" in logs
-        assert "autoscaler" in logs
-
-        # Test glob works.
-        logs = list_logs(node_id=head_node["node_id"], glob_filter="raylet*")
-        assert len(logs) == 1
-        return True
+        for node in list_nodes():
+            # When glob filter is not provided, it should provide all logs
+            logs = list_logs(node_id=node["node_id"])
+            assert "raylet" in logs
+            assert "gcs_server" in logs
```

Comment:
gcs_server is not in worker nodes. You probably need to separate out head node & worker nodes?